diff --git a/ogr2ogr.class.inc b/ogr2ogr.class.inc
index 64b6315..5fa4852 100644
--- a/ogr2ogr.class.inc
+++ b/ogr2ogr.class.inc
@@ -22,7 +22,7 @@ class Ogr2ogr {
       if ($this->uri) {
         $result = array();
 
-        $command = $this->ogr2ogr .' -f CSV /vsistdout/ "'. $this->uri .'"';
+        $command = $this->ogr2ogr .' -f CSV /vsistdout/ "'. $this->uri .'" -dim 2';
         foreach ($args as $flag => $value) {
           $command .= ' '. $flag .' '. $value;
         }
